### PR TITLE
[WIP] Require password to change email address

### PR DESCRIPTION
**Pull request by eessex**
The type of this PR is: feature

This PR resolves [PLATFORM-3026]

### Description

Adds a required password field to the form for changing a user's email address.

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [ ] I have documented any follow-up work that this PR will require, or it does not require any.
- [ ] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [ ] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[PLATFORM-3026]: https://artsyproduct.atlassian.net/browse/PLATFORM-3026

### CODE REVIEW

File: src/lib/Scenes/MyAccount/updateMyUserProfile.ts [hunk 1]
```diff
@@ -5,7 +5,12 @@ import { commitMutation, graphql } from "react-relay"
...
     commitMutation<updateMyUserProfileMutation>(defaultEnvironment, {
-      onCompleted: resolve,
+      onCompleted: (response, _) => {
+        if (response.updateMyUserProfile?.userOrError?.mutationError) {
+          reject(response.updateMyUserProfile?.userOrError?.mutationError)
+        }
+        resolve(response)
+      },
       mutation: graphql`
         mutation updateMyUserProfileMutation($input: UpdateMyProfileInput!) {
```
Comment (joeyAghion) [inline src/lib/Scenes/MyAccount/updateMyUserProfile.ts:15]: Sorry for the noob question, but where is the new password actually passing through to the back end? Is it in this `input`?

**Review by joeyAghion — COMMENTED** [review]
In API logs, Eigen's user-updating requests appear with parameters passed in _querystrings_. Can you confirm that Metaphysics is responsible for that, and that fix can be made independently of this PR? (We don't want sensitive parameters appearing in querystrings, so they're not recorded in logs and elsewhere.)

File: src/lib/Scenes/MyAccount/MyAccountEditEmail.tsx [hunk 2]
```diff
@@ -29,10 +38,18 @@ const MyAccountEditEmail: React.FC<{ me: MyAccountEditEmail_me; relay: RelayProp
       canSave={isEmailValid}
       onSave={async (dismiss) => {
         try {
-          await updateMyUserProfile({ email })
+          await updateMyUserProfile({ email, password })
```
Comment (eessex) [inline src/lib/Scenes/MyAccount/MyAccountEditEmail.tsx:41]: @joeyAghion this is where we pass `$input` to metaphysics

**Comment by brainbicycle** [comment]
We need to find a way to handle case of 3rd party logins without passwords before we can merge this, see thread here:
https://artsy.slack.com/archives/CA8SANW3W/p1611763435135500

**Comment by joeyAghion** [comment]
One thought about this: we anticipate requiring password for more operations in the future (such as managing 2FA or deleting an account), so if there's a way to generalize this functionality, that could be worthwhile.

**Comment by MounirDhahri** [comment]
@joeyAghion @brainbicycle this has been inactive for a while, should we prioritize this?

**Pull request closed without merging**